src/nm.c

```diff
@@ -344,6 +344,9 @@ write_wireguard_params(const NetplanNetDefinition* def, GKeyFile *kf, GError** e
             g_key_file_set_string(kf, "wireguard", "private-key", def->tunnel.private_key);
     }
 
+    if (def->backend_settings.private_key_flags != NETPLAN_PRIVATE_KEY_FLAG_NONE)
+        g_key_file_set_uint64(kf, "wireguard", "private-key-flags", def->backend_settings.private_key_flags);
```

Comment:
thought: I wonder if the same data-structure could be used for `[wifi-security].psk-flags`?

Reply:
If only wireguard and the the psk password will have flags we could use the same variable, as the netdef will never be wifi and wireguard at the same time. But if we want to extend it to all the other secrets we will need multiple variables...

The case for wifi é covered by the passthrough data, only the wireguard one was causing issues due to how NM parses the keyfile and how we were generating it. Do we want to extend it to wifi right now?